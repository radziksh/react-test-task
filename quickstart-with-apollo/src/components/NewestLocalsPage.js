import React, { Fragment } from 'react'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'
import { Col, Row } from 'react-bootstrap'
import TipComponent from './PlaceComponent'

class NewestLocalsPage extends React.Component {

  componentWillReceiveProps(nextProps) {
    if (this.props.location.key !== nextProps.location.key) {
      this.props.allLocalsQuery.refetch()
    }
  }

  render() {
    if (this.props.allLocalsQuery.loading) {
      return (
        <div className='flex w-100 h-100 items-center justify-center pt7'>
          <div>
            Loading...         
          </div>
        </div>
      )
    }

    return (
      <Fragment>
        {this.props.allLocalsQuery.allLocals && this.props.allLocalsQuery.allLocals.map(post => (
          <TipComponent
            key={post.id}
            imageUrl={post.imageUrl}
            description={post.name}
            refresh={() => this.props.allLocalsQuery.refetch()}
          />
        ))}
        {this.props.children}
      </Fragment>
    )
  }
}

const ALL_LOCALS_QUERY = gql`
  query allLocalsQuery {
    allLocals(orderBy: createdAt_DESC, filter:{
      verified: false
    }) {
      id
      imageUrl
      name
    }
  }
`

const NewestLocalsPageWithQuery = graphql(ALL_LOCALS_QUERY, {
  name: 'allLocalsQuery',
  options: {
    fetchPolicy: 'network-only',
  },
})(NewestLocalsPage)

export default NewestLocalsPageWithQuery
