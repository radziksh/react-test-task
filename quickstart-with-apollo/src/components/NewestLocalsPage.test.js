
/**
 * @jest-environment node
 */
import React from 'react'
import { NewestLocalsPage } from './NewestLocalsPage'
test('render NewestLocalsPage', () => {
  const wrapper = global.render(
    <NewestLocalsPage allLocalsQuery={{
      loading: false
    }}
    />
  )
  expect(wrapper).toMatchSnapshot()
})

test('render NewestLocalsPage with loading state', () => {
  const wrapper = global.render(
    <NewestLocalsPage allLocalsQuery={{
      loading: true
    }}
    />
  )
  expect(wrapper).toMatchSnapshot()
})
