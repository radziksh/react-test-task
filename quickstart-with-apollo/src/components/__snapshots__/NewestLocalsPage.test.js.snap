// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render NewestLocalsPage 1`] = `null`;

exports[`render NewestLocalsPage with loading state 1`] = `
<div
  class="flex w-100 h-100 items-center justify-center pt7"
>
  <div>
    Loading...
  </div>
</div>
`;
